src/storage/datacore/restapi/mode/alertscount.pm

```diff
@@ -0,0 +1,178 @@
+# limitations under the License.
+#
+package storage::datacore::restapi::mode::alertscount;
+use strict;
+use warnings;
+use centreon::plugins::misc qw(empty);
+
+use base qw(centreon::plugins::templates::counter);
+
+my %alerts_level = ('trace' => 0, 'info' => 1, 'warning' => 2, 'error' => 3);
+
+sub new {
+    my ($class, %options) = @_;
+
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+    $options{options}->add_options(arguments => {
+        'filter-server:s' => { name => 'filter_server' },
+        'max-alert-age:s' => { name => 'max_alert_age' } });
+
+    $self->{output} = $options{output};
+
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'alerts', type => 0 },
+    ];
+    $self->{maps_counters}->{alerts} = [
+        {
+            label  => 'error',
+            nlabel => 'datacore.event.error.count',
+            set    => {
+                key_values      => [ { name => 'error' } ],
+                output_template => 'number of error alerts : %s',
+                perfdatas       => [ { template => '%d', min => 0 } ]
+            }
+        }, {
+        label  => 'warning',
+        nlabel => 'datacore.alerts.warning.count',
+        set    => {
+            key_values      => [ { name => 'warning' } ],
+            output_template => 'number of warning alerts : %s',
+            perfdatas       => [ { template => '%d', min => 0 } ]
+        }
+    }, {
+        label  => 'info',
+        nlabel => 'datacore.alerts.info.count',
+        set    => {
+            key_values      => [ { name => 'info' } ],
+            output_template => 'number of info alerts : %s',
+            perfdatas       => [ { template => '%d', min => 0 } ]
+        }
+    }, {
+        label  => 'trace',
+        nlabel => 'datacore.alerts.trace.count',
+        set    => {
+            key_values      => [ { name => 'trace' } ],
+            output_template => 'number of trace alerts : %s',
+            perfdatas       => [ { template => '%d', min => 0 } ]
+        }
+    },
+    ];
+
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $data = $options{custom}->request_api(
+        url_path => '/RestService/rest.svc/1.0/alerts'
+    );
+    my $alerts_count = $self->order_alerts($data);
+    $self->{alerts} = {
+        trace   => $alerts_count->{$alerts_level{trace}}->{count},
+        info    => $alerts_count->{$alerts_level{info}}->{count},
+        warning => $alerts_count->{$alerts_level{warning}}->{count},
+        error   => $alerts_count->{$alerts_level{error}}->{count},
+    };
+
+}
+
+# take a decoded json reference and send back a hash with loglevel as key and a hash with number of alert
+# and the list of alerts text as value.
+sub order_alerts {
```

Comment:
I'm not sure to understand why you put a part of manage_selection in another function while usually all this stuff is done in a loop core like [here](https://github.com/centreon/centreon-plugins/blob/c245c4104160636d900256ba0abd1bc836437467/src/network/cisco/vcs/restapi/mode/alerts.pm#L91)